websock/http/server.nim

```diff
@@ -191,22 +192,36 @@ proc accept*(server: HttpServer): Future[HttpRequest]
 
   trace "Got new request", isTls = server.secure
   try:
-    return await server.parseRequest(stream)
+    let parsedRequestfut = server.parseRequest(stream)
+    let res = await withTimeout(parsedRequestfut, server.handshakeTimeout)
+    if not res:
+      # Timeout
+      raise newException(HttpError, "Didn't Parse request in time!")
```

Comment:
I'd rather we don't raise multiple times in this way. Can we restructure this so that we only raise once with the appropriate error?